### PR TITLE
fix: Improve Lineage chart UI

### PR DESCRIPTION
**Pull request by ozandogrultan**
Signed-off-by: Ozan Dogrultan <ozan.dogrultan@deliveryhero.com>

### Summary of Changes

This PR introduces some UI improvements to especially fix the overlapping of node labels in the Lineage chart when there are many nodes being rendered.

This is achieved by assigning a fixed the horizontal space between different labels in the graph and rearranging the position of the labels based on the graph direction.

Here are the screenshot to compare before and after:

*Before:*
Screenshot 2022-04-15 at 12 02 43

*After:*
Screenshot 2022-04-15 at 11 10 04

### Tests

Updated one test case for Lineage graph dimension calculation.

### Documentation

N/A

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 1]
```diff
@@ -333,8 +340,19 @@ export const buildNodes = (g, targetNode, nodes, onClick) => {
   // Position node label
   nodeEnter
     .append('text')
-    .attr('dy', NODE_LABEL_Y_OFFSET)
-    .attr('text-anchor', 'middle')
+    .attr('x', (d) => (d.y < 0 ? NODE_LABEL_X_OFFSET : -NODE_LABEL_X_OFFSET))
+    .attr('dy', (d) =>
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts:344]: maybe getLabelYOffset?

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 2]
```diff
@@ -461,7 +479,10 @@ export const buildSVG = (el: HTMLElement, dimensions: Dimensions) => {
     .append('svg')
     .classed('svg-content', true)
     .attr('width', dimensions.width)
-    .attr('height', dimensions.height);
+    .attr('height', dimensions.height)
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts:482]: Duplicate

**Review by Golodhros — APPROVED** [review]
LGTM

Some minor comments.

File: frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts [hunk 3]
```diff
@@ -318,6 +320,11 @@ export const buildNodes = (g, targetNode, nodes, onClick) => {
     // eslint-disable-next-line no-return-assign
     .data(nodes, ({ id }) => id);
 
+  // Normalize for fixed-depth.
+  nodes.forEach((d) => {
+    d.y = d.y < 0 ? d.depth * -NODE_WIDTH : d.depth * NODE_WIDTH;
+  });
```
[the remark this answers is not in the document]
Comment (ozandogrultan) [reply frontend/amundsen_application/static/js/components/Lineage/Graph/chart.ts]: I've moved this logic to the existing `decompactLineage` method

**Comment by ozandogrultan** [comment]
Hi @Golodhros , I've addressed your comments, can you please take a look again? Thanks a lot!

**Review by feng-tao — APPROVED** [review]
thanks for improving the product!

**Pull request merged**